src/app/components/CoursesContainer.tsx

```diff
@@ -1,28 +1,20 @@
 
-import React from "react";
-import Link from "next/link";
+import { CourseCard } from "./CourseCard";
 
 
 const CoursesContainer =  ({courses}: any) => {
 
 
   return (
     <>
-      <div className="">
-        <ul className="grid grid-cols-1 gap-4 md:grid-cols-3 my-3 place-items-center p-3">
+      <div className="grid sm:grid-cols-2 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 2xl:grid-cols-4 gap-4 my-5 mx-5">
           {courses?.map((course: any) => {
             return (
-              <div key={course.courseCode} className=" bg-blue-100 h-32 p-3 w-full max-w-96 md:max-w-96 m-2 rounded place-items-center hover:shadow-lg hover:shadow-yellow-400">
-                  <Link href={`/courses/${course.courseCode}`}>
-                    <div className="flex flex-col items-center justify-center text-blue-700">
-                      <p className="text-black font-bold">{course.courseCode}</p>
-                      <p>{course.title}</p>
-                    </div>
-                  </Link>
+              <div key={course.courseCode} className="">
+                  <CourseCard courseCode={course.courseCode} title={course.title} imageUrl={course.imageUrl} courseType={course.courseType} courseCategory={course.courseCategory} />
```

Comment:
For thumbnail, requires an `imageUrl` field on the API. Using placeholder image on courseCard.tsx for now

Reply:
also, instead of `imageUrl`, what about implementing something like  adding a text `course code` in front of the `card image` ? It's doable I think.

Reply:
> also, instead of `imageUrl`, what about implementing something like adding a text `course code` in front of the `card image` ? It's doable I think.

yes sure, seems doable so I'll look into it further. Also I'll try fixing the margins and grid gaps along with max-width so it fits better and send another PR sometime soon.